doc/api/esm.md

```diff
@@ -1130,18 +1130,40 @@ node --experimental-top-level-await b.mjs # works
 <!-- type=misc -->
 
 To customize the default module resolution, loader hooks can optionally be
-provided via a `--experimental-loader ./loader-name.mjs` argument to Node.js.
-
+provided via `--experimental-loader ./loader-name.mjs` arguments to Node.js.
 When hooks are used they only apply to ES module loading and not to any
 CommonJS modules loaded.
 
+Loaders are modules which export functions and data which can be used by
+Node.js to modify the behavior of loading ES Modules. Loaders may be “chained”
+together by passing multiple `--experimental-loader` arguments to Node.js. A
+loader module will be imported once as a singleton.
+
+Loaders follow an authority model, whereby each loader passed to node, from
+left to right, takes precedence over the next. For the `resolve`, `getFormat`,
+and `getSource` hooks, this means that the leftmost hook is called, and then
+the next, and so on, until one gives an answer. For `transformSource`, this
+means the rightmost hook is called first, and the result is passed to the next,
+and so on, until it reaches the leftmost hook. The `getGlobalPreloadCode` hooks
+run from left to right, so that the loaders to the left have more control over
+the environment than loaders to the right.
```

Comment:
One of my main outstanding conerns is the directionality of the loader hooks, which is what I was trying to express in https://github.com/nodejs/node/pull/33812#issuecomment-650667286 by using a concrete example. It does not make logical sense for `transformSource` to be called from right to left and `resolve` to be called from left to right. 

Take the case where I am chaining the `resolve` hook to do the progressive resolution of module specifiers. How is this different from progressive transpilation of the module source text that happens with `transformSource`? Having these two hooks be called in opposite directions seems unintuitive to me.

So, rather than adding prose to help clarify this unintuitive behavior, why not make it function as one would expect? That is, having all the hooks be called in the same order?

Reply:
The functions are called such that the leftmost loader has the most authority. reversing the order that the transform hooks are called in would break that.

Reply:
The authority of a loader module need not be reflected in the direction in which its hooks are called as it's bad UX and provides a false sense of security. The leftmost loader has the most authority because it is executed first, not because it has the final say on what the transformed source code should be. Because the leftmost loader has the highest authority (executed first), it can even delete the files of the subsequent loaders if it wanted to. 

The following is how I would expect the UX of chained loaders to be.

```bash
node ./entrypoint.mjs --loader=./transform1.mjs --loader=./transform2.mjs --loader=./transform3.mjs
                               ^^^^^^^^^^^^^^^^                                    ^^^^^^^^^^^^^^^^
                               Initial transform                                   Final transform
```

Reply:
that doesn't work. transform1 needs final say over the transformation.

Reply:
> that doesn't work. transform1 needs final say over the transformation.

Do you want to justify that?

Reply:
> The leftmost loader has the most authority because it is executed first

I think that is an awkward use of "authority". Authority in this context is usually about controlling the final behavior. So if the last thing to run controls the final behavior (as is the case with serial-chained transform), then the last thing to run has the most authority. It's about order-of-return, not order-of-entry.

We shouldn't reuse authority to also mean "order of execution" because imo that's just confusing.

Reply:
> So, if the last thing to run controls the final behavior (as is the case with serial-chained transform), then the last thing to run has the most authority.

SGTM, and using that definition, my concern about the unintuitive directionality would be resolved. This would result in good UX as all hooks would be called in the same direction (left to right in this case).